src/samples/PoseWithCovariance.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+#define BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a 3D pose with uncertainty of the frame 'object_frame_id' in the frame 'frame_id'.
+ * A frame ID shall be globally unique and correspond to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The pose \f$ ^{reference}_{object}T \f$ in this structure transforms a vector \f$ v_{object} \f$
+ * from the object frame to the reference frame:
+ * \f$ v_{reference} = ^{reference}_{object}T \cdot v_{object} \f$
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
```

Comment:
>  I don't see much benefit in forwarding all those methods.

Me neither. At this stage, I would really much rather prefer keeping the interface simple, as it was originally.

As for the frame checks, it's IMO a bad idea to start injecting these kind of checks in a type like this, which is not meant for general computation (again, AFAIK). If one wants proper typecheck-level frame checking, then one should go and try KUL's semantic geometry (or, at least, attempt at doing a clean compile-time implementation of it that does not entail doing string comparisons all the time).